consent/api/v1/views.py

```diff
@@ -232,7 +280,12 @@ def delete(self, request):
         user_id = get_user_id(username)
         self.set_consent_state(False, user_id, course_id, enterprise_customer_uuid)
         provided = consent_provided(user_id, course_id, enterprise_customer_uuid)
-        required = consent_required(user_id, course_id, enterprise_customer_uuid)
+        enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+        if enterprise_customer and not enterprise_customer.catalog_contains_course_run(request.user, course_id):
+            required = False
```

Comment:
@haikuginger Let's add the comment from L182-L283 here as well.

> Only mark consent as required if it's in the Enterprise-linked course catalog; not otherwise.